cwl/src/test/scala/wdl4s/cwl/ThreeStepExample.scala

```diff
@@ -0,0 +1,148 @@
+package wdl4s.cwl
+
+import shapeless._
+import syntax.singleton._
+import wdl4s.cwl.CommandLineTool.{Argument, BaseCommand, Inputs, StringOrExpression}
+import wdl4s.cwl.CommandOutputBinding.Glob
+import wdl4s.cwl.WorkflowStep.{Outputs, Run}
+import io.circe.syntax._
+import io.circe.yaml._
+import io.circe.yaml.syntax._
+import org.scalatest._
+
+/*
+ This example calls `ps` , then counts the number of processes that match a pattern input.
+ The output of the ps call is also counted via `wc`.
+ */
+object ThreeStepExample extends App {
```

Comment:
I guess I think it'd be valuable to check, for a bunch of different CWL files and features:

- The parser can read the file into CWL4S
- The CWL4S can be written back to JSON
- The JSON output is equivalent to the JSON input